controllers/terminal_controller.go

```diff
@@ -724,16 +732,34 @@ func deleteClusterRoleBinding(ctx context.Context, cs *ClientSet, name string) e
 	return deleteObj(ctx, cs, clusterRoleBinding)
 }
 
-func (r *TerminalReconciler) createOrUpdateAdminKubeconfig(ctx context.Context, targetClientSet *ClientSet, hostClientSet *ClientSet, t *extensionsv1alpha1.Terminal, labelsSet *labels.Set, annotationsSet *utils.Set) (*corev1.Secret, error) {
-	accessSecret, err := r.createAccessToken(ctx, targetClientSet, t, labelsSet, annotationsSet)
+type adminKubeconfig struct {
+	kubeconfig corev1.Secret
+	token      corev1.Secret
+}
+
+func (r *TerminalReconciler) createOrUpdateAdminKubeconfig(ctx context.Context, targetClientSet *ClientSet, hostClientSet *ClientSet, t *extensionsv1alpha1.Terminal, labelsSet *labels.Set, annotationsSet *utils.Set) (*adminKubeconfig, error) {
+	accessToken, err := r.createAccessToken(ctx, targetClientSet, t, labelsSet, annotationsSet)
 	if err != nil {
 		return nil, err
 	}
 
-	return createOrUpdateKubeconfig(ctx, targetClientSet, hostClientSet, t, accessSecret, labelsSet, annotationsSet)
+	kubeconfig, err := createOrUpdateKubeconfig(ctx, targetClientSet, hostClientSet, t, labelsSet, annotationsSet)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create or update kubeconfig secret: %w", err)
+	}
+
+	token, err := createOrUpdateToken(ctx, hostClientSet, t, accessToken, labelsSet, annotationsSet)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create or update token secret: %w", err)
+	}
+
+	return &adminKubeconfig{
+		kubeconfig: *kubeconfig,
+		token:      *token,
+	}, nil
 }
 
-func (r *TerminalReconciler) createAccessToken(ctx context.Context, targetClientSet *ClientSet, t *extensionsv1alpha1.Terminal, labelsSet *labels.Set, annotationsSet *utils.Set) (*corev1.Secret, error) {
+func (r *TerminalReconciler) createAccessToken(ctx context.Context, targetClientSet *ClientSet, t *extensionsv1alpha1.Terminal, labelsSet *labels.Set, annotationsSet *utils.Set) (*authenticationv1.TokenRequestStatus, error) {
```

Comment:
rename to something like `createOrUpdateAccessServiceAccountAndRequestToken`

Reply:
✅